server/utils/dateUtils.ts

```diff
@@ -269,3 +269,10 @@ export const getTodaysDatePlusMonthsAndDays = (monthsToAdd = 0, daysToAdd = 0):
 }
 
 export class InvalidDateStringError extends Error {}
+
+export const dateIsBlank = <K extends string | number>(
+  dateInputObj: Partial<ObjectWithDateParts<K>>,
+  key: K,
+): boolean => {
+  return !['year' as const, 'month' as const, 'day' as const].every(part => !!dateInputObj[`${key}-${part}`])
```

Comment:
Suggestion only: There's quite a bit happening here on a single line, especially when using `!`s to invert things.

I'd personally lean towards something more verbose for more readability, something like:

```
const dateParts = ['year', 'month', 'day'] as const;
let allPartsExist = true;

for (const part of dateParts) {
    const keyExists = Boolean(dateInputObj[`${key}-${part}`]);
    if (!keyExists) {
        allPartsExist = false;
        break;
    }
}
```

Reply:
yes, agree this is clearer. I've amended it 👍